NitroxPatcher/Patches/SubNameInput_OnColorChange_Patch.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Harmony;
+using NitroxClient.GameLogic.Helper;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using System;
+using System.Reflection;
+
+namespace NitroxPatcher.Patches
+{
+    public class SubNameInput_OnColorChange_Patch : NitroxPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(SubNameInput);
+        public static readonly MethodInfo TARGET_METHOD = TARGET_CLASS.GetMethod("OnDeselect", BindingFlags.Public | BindingFlags.Instance);
+        public static readonly MethodInfo TARGET_METHOD2 = TARGET_CLASS.GetMethod("SetSelected", BindingFlags.Public | BindingFlags.Instance);
+
+        public static void Postfix(SubNameInput __instance)
+        {
+            int index = __instance.SelectedColorIndex;
+            SubName subname = (SubName)__instance.ReflectionGet("target");
+            if (Player.main.GetCurrentSub() != null && subname != null)
+            {
+                SubNameInput.ColorData[] colorData = __instance.colorData;
+                if (colorData != null && index < colorData.Length)
+                {
+                    UnityEngine.Color color = colorData[index].image.color;
+                    UnityEngine.Vector3 hsb = subname.GetColor(index);
+                    Color colorDataColor = new Color(color.r, color.g, color.b, color.a);
+                    Vector3 colorDataHSB = new Vector3(hsb.x, hsb.y, hsb.z);
+                    String guid = GuidHelper.GetGuid(Player.main.GetCurrentSub().gameObject);
+                    Multiplayer.Logic.Cyclops.ChangeColor(guid, __instance.SelectedColorIndex, colorDataHSB, colorDataColor);
+                }
+            }
+        }
+
+        public static void Prefix(SubNameInput __instance)
```

Comment:
From the looks of it `Postfix` and `Prefix` are identical, so it's better to move their contents into a separate function and call that.

There should also be some functions (or I added them on my own branch) to specify which function to use for patching, in that case you can have only one function, and call `this.Patch*(harmony, TARGET_METHOD*, patch_method)`. I'll update this answer once I know for sure.